src/main/java/com/epam/page/object/generator/validators/AbstractValidator.java

```diff
@@ -66,12 +75,21 @@ public void validate(ValidationContext validationContext) {
         searchRules.addAll(isValidateAllSearchRules ? validationContext.getAllSearchRules()
             : validationContext.getValidRules());
 
-        searchRules.forEach(searchRule -> {
-            validationContext
-                .addValidationResult(
-                    !isValid(searchRule, validationContext) ? new ValidationResult(false, this,
-                        searchRule) : new ValidationResult(true, this, searchRule));
-        });
+        searchRules.stream()
+            .filter(searchRule -> supportedSearchRuleTypes != null
+                && (supportedSearchRuleTypes.contains(SearchRuleType.ALL)
+                        || supportedSearchRuleTypes.stream()
+                                .anyMatch(searchRuleType ->
+                                    searchRuleType.getName().equals(searchRule.getType()))
+                    )
```

Comment:
6-lines lambda is a bad style. I suggest extracting a method.

Reply:
Done